pkg/controller/clusterconfig/migration.go

```diff
@@ -0,0 +1,189 @@
+		// In live migration, preserve the network type change
+		operConfig.Spec.DefaultNetwork.Type = currentOperConfig.Spec.DefaultNetwork.Type
+
+		if meta.IsStatusConditionPresentAndEqual(clusterConfig.Status.Conditions, names.NetworkTypeMigrationInProgress, metav1.ConditionTrue) {
+			if v1helpers.IsOperatorConditionTrue(operConfig.Status.Conditions, operv1.OperatorStatusTypeProgressing) {
+				// Not update network.operator if the operator is processing
+				return nil
+			}
+			operConfig.Spec.Migration = currentOperConfig.Spec.Migration
+			err = r.prepareOperatorConfigForNetworkTypeMigration(ctx, clusterConfig, operConfig)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func (r *ReconcileClusterConfig) readMTUConfigMap(ctx context.Context) (int, error) {
+	cmNamespace := "openshift-network-operator"
+	cmName := "mtu"
+	log.Printf("Looking for ConfigMap %s/%s", cmNamespace, cmName)
+	cm := &corev1.ConfigMap{}
+	err := r.client.Default().CRClient().Get(ctx, types.NamespacedName{Namespace: cmNamespace, Name: cmName}, cm)
+	if err != nil {
+		return 0, err
+	}
+	mtu, err := strconv.Atoi(cm.Data["mtu"])
+	if err != nil || mtu == 0 {
+		return 0, fmt.Errorf("format error")
+	}
+
+	log.Printf("Found mtu %d", mtu)
+	return mtu, nil
+}
+
+// GetNetworkCondition extracts the provided condition from the given status and returns that.
+// Returns nil and -1 if the condition is not present, and the index of the located condition.
+func GetNetworkCondition(status *configv1.NetworkStatus, conditionType string) (int, *metav1.Condition) {
+	if status == nil {
+		return -1, nil
+	}
+	for i := range status.Conditions {
+		if status.Conditions[i].Type == conditionType {
+			return i, &status.Conditions[i]
+		}
+	}
+	return -1, nil
+}
+
+func (r *ReconcileClusterConfig) prepareOperatorConfigForNetworkTypeMigration(ctx context.Context, clusterConfig *configv1.Network, operConfig *operv1.Network) error {
+	status := clusterConfig.Status
+	if status.Conditions == nil {
+		return fmt.Errorf("status.Conditions is not initialized")
+	}
+
+	mcpCondition := meta.FindStatusCondition(status.Conditions, names.NetworkTypeMigrationMTUReady)
+	if mcpCondition == nil {
+		return fmt.Errorf("condition %q not found", names.NetworkTypeMigrationMTUReady)
+	}
+	if mcpCondition.Reason == names.MachineConfigPoolsUpdating {
+		log.Printf("MCP is updating, so we don't modify the operator config")
+		return nil
+	}
+
+	mtuApplied := meta.IsStatusConditionPresentAndEqual(status.Conditions, names.NetworkTypeMigrationMTUReady, metav1.ConditionTrue)
+	cniReady := meta.IsStatusConditionPresentAndEqual(status.Conditions, names.NetworkTypeMigrationTargetCNIAvailable, metav1.ConditionTrue)
+	mcApplied := meta.IsStatusConditionPresentAndEqual(status.Conditions, names.NetworkTypeMigrationTargetCNIInUse, metav1.ConditionTrue)
+
+	var isRollback bool
+	if clusterConfig.Spec.NetworkType == "OpenShiftSDN" {
```

Comment:
The main difference is when we apply the routeble MTU. For migration, routable MTU is applied before switching the nodes CNI to OVN-K. For rollback, the routable MTU is applied togther with switching the nodes CNI to OVN-K. When a node is rebooted and SDN is used as the node CNI. So that pods on SDN nodes wouldn't send packets with MTU larger than that cannot be handled by OVN-K.